### ISSUE TITLE
About the value of vmax in chapter 3 (For instance: code In[74], page 198)

### ISSUE THREAD

**Issue by gaussEuler** [amueller/introduction_to_ml_with_python#55]
Respected Dr. Muller,

When I ran the code of In[74] (page 198), I got yellow boxes without any image! 
Then I found that changing vmax from 1 to ~200 produces proper images as shown in the book in page 199. (Changing vmin had no drastic effect). 
I think vmax corresponds to contrast while vmin corresponds to brightness (am I right?)
(I am using Spyder 3.2.4 (with Python 3.5.4, 64 bits on Windows 10).

**Comment by gaussEuler** [comment]
The code (exactly as in the book) is:
import numpy as np
import matplotlib.pyplot as plt

from sklearn.datasets import fetch_lfw_people
people = fetch_lfw_people(min_faces_per_person=20, resize=0.7)

mask = np.zeros(people.target.shape, dtype=np.bool)
for target in np.unique(people.target):
    mask[np.where(people.target == target)[0][:50]] = 1    
X_people = people.data[mask]
y_people = people.target[mask]

from sklearn.decomposition import PCA
pca = PCA(n_components=100, whiten=True, random_state=0)
pca.fit_transform(X_people)
X_pca = pca.transform(X_people)

from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
scaler.fit(X_people)
X_scaled = scaler.transform(X_people)

image_shape = people.images[0].shape

from sklearn.cluster import DBSCAN
dbscan = DBSCAN(min_samples=3, eps=15)
labels = dbscan.fit_predict(X_pca)

noise = X_people[labels==-1]

fig, axes = plt.subplots(3, 9, subplot_kw={'xticks': (), 'yticks': ()},
                         figsize=(12, 4))
for image, ax in zip(noise, axes.ravel()):
    ax.imshow(image.reshape(image_shape), vmin=0, vmax=1)

**Comment by gaussEuler** [comment]
vmax value is mentioned in the last line of the above code

**Comment by gaussEuler** [comment]
I think that vmax=1 should be around vmax=200

**Comment by amueller** [comment]
Thanks for opening the issue and sorry you're having trouble. What versions of scikit-learn and matplotlib are you using?

**Comment by amueller** [comment]
I can reproduce the behavior copy and pasting your code. Let me see what's different from the notebook in this repository.

**Comment by gaussEuler** [comment]
The scikit-learn version is 0.19.1
The matplotlib version is 2.1.0

**Comment by amueller** [comment]
There's this line in the book that you're missing:
```python
X_people = X_people / 255.
```

**Comment by amueller** [comment]
I also set
```python
plt.rcParams['image.cmap'] = "gray"
```
which will make it black and white instead of using the green-yellow colormap (for all following plots).

**Comment by gaussEuler** [comment]
Yes, using X_people = X_people / 255, I am able to correctly get the images using vmax=1
Thank you so much sir. I feel really stupid!

**Comment by amueller** [comment]
No worries. Overwriting a variable is probably not the best style.

**Comment by gaussEuler** [comment]
Thank you again so much for your prompt reply in correcting my mistake. I sincerely appreciate.
